src/msg/async/rdma/RDMAConnectedSocketImpl.h

```diff
@@ -78,35 +81,19 @@ class RDMAConnectedSocketImpl : public ConnectedSocketImpl {
   virtual int fd() const override { return notify_fd; }
   void fault();
   const char* get_qp_state() { return Infiniband::qp_state_string(qp->get_state()); }
-  ssize_t submit(bool more);
-  int activate();
-  void fin();
-  void handle_connection();
-  void cleanup();
-  void set_accept_fd(int sd);
-  int try_connect(const entity_addr_t&, const SocketOptions &opt);
-
-  class C_handle_connection : public EventCallback {
-    RDMAConnectedSocketImpl *csi;
-    bool active;
-   public:
-    C_handle_connection(RDMAConnectedSocketImpl *w): csi(w), active(true) {}
-    void do_request(int fd) {
-      if (active)
-        csi->handle_connection();
-    }
-    void close() {
-      active = false;
-    }
-  };
+  virtual void fin();
+  virtual void cleanup() = 0;
+  virtual int try_connect(const entity_addr_t&, const SocketOptions &opt) = 0;
+  virtual ssize_t submit(bool more);
+  virtual int remote_qpn() = 0;
```

Comment:
is this worth to define a virtual interface? I think we could have a variable in RDMAConnectedSocketImpl